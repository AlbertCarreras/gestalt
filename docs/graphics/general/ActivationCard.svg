<svg width="325" height="180" viewBox="0 0 325 180" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="325" height="180" fill="#F6FDF5"/>
<path d="M46 28C46 23.5817 49.5817 20 54 20H271C275.418 20 279 23.5817 279 28V152C279 156.418 275.418 160 271 160H54C49.5817 160 46 156.418 46 152V28Z" fill="white"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M271 22H54C50.6863 22 48 24.6863 48 28V152C48 155.314 50.6863 158 54 158H271C274.314 158 277 155.314 277 152V28C277 24.6863 274.314 22 271 22ZM54 20C49.5817 20 46 23.5817 46 28V152C46 156.418 49.5817 160 54 160H271C275.418 160 279 156.418 279 152V28C279 23.5817 275.418 20 271 20H54Z" fill="#005C62"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M70.2699 40.5611L69.2794 43.9584C69.1966 44.2361 68.8034 44.2355 68.7206 43.9578L67.7295 40.5611H67.7347C67.7067 40.4543 67.6875 40.3447 67.6875 40.2292C67.6875 39.5041 68.2749 38.9167 69 38.9167C69.7245 38.9167 70.3125 39.5041 70.3125 40.2292C70.3125 40.3447 70.2927 40.4543 70.2647 40.5611H70.2699ZM69 47.0833C68.4359 47.0833 67.9792 46.626 67.9792 46.0625C67.9792 45.4984 68.4359 45.0417 69 45.0417C69.5635 45.0417 70.0208 45.4984 70.0208 46.0625C70.0208 46.626 69.5635 47.0833 69 47.0833ZM69 36C65.1337 36 62 39.1337 62 43C62 46.8657 65.1337 50 69 50C72.8657 50 76 46.8657 76 43C76 39.1337 72.8657 36 69 36Z" fill="#005C62"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M257.856 42.9998L262.032 38.8237C262.545 38.311 262.545 37.4803 262.032 36.9676C261.52 36.4548 260.688 36.4548 260.176 36.9676L256 41.1437L251.824 36.9676C251.311 36.4548 250.48 36.4548 249.968 36.9676C249.455 37.4803 249.455 38.311 249.968 38.8237L254.144 42.9998L249.968 47.1759C249.455 47.6887 249.455 48.5193 249.968 49.0321C250.224 49.2882 250.56 49.4165 250.896 49.4165C251.232 49.4165 251.568 49.2882 251.824 49.0321L256 44.856L260.176 49.0321C260.432 49.2882 260.768 49.4165 261.104 49.4165C261.44 49.4165 261.776 49.2882 262.032 49.0321C262.545 48.5193 262.545 47.6887 262.032 47.1759L257.856 42.9998Z" fill="#005C62"/>
<rect x="65" y="120" width="195" height="28" rx="12.5625" fill="#005C62"/>
<path d="M149.401 138.848C150.791 138.848 151.653 138.089 151.653 136.876C151.653 135.988 151.025 135.305 150.163 135.234V135.148C150.831 135.043 151.341 134.408 151.341 133.677C151.341 132.612 150.588 131.958 149.324 131.958H146.477V138.848H149.401ZM147.839 133.028H148.977C149.626 133.028 150.001 133.362 150.001 133.925C150.001 134.498 149.599 134.823 148.864 134.823H147.839V133.028ZM147.839 137.774V135.773H149.022C149.825 135.773 150.263 136.117 150.263 136.762C150.263 137.42 149.838 137.774 149.053 137.774H147.839Z" fill="white"/>
<path d="M157.257 133.591H155.944V136.628C155.944 137.349 155.592 137.793 154.938 137.793C154.333 137.793 154.022 137.416 154.022 136.671V133.591H152.709V137.015C152.709 138.237 153.376 138.958 154.45 138.958C155.204 138.958 155.669 138.614 155.903 138.003H155.985V138.848H157.257V133.591Z" fill="white"/>
<path d="M158.949 132.335V133.639H158.173V134.694H158.949V137.43C158.949 138.452 159.428 138.867 160.637 138.867C160.89 138.867 161.134 138.839 161.296 138.805V137.779C161.17 137.793 161.079 137.802 160.908 137.802C160.461 137.802 160.263 137.583 160.263 137.11V134.694H161.296V133.639H160.263V132.335H158.949Z" fill="white"/>
<path d="M162.812 132.335V133.639H162.036V134.694H162.812V137.43C162.812 138.452 163.291 138.867 164.5 138.867C164.753 138.867 164.996 138.839 165.159 138.805V137.779C165.033 137.793 164.942 137.802 164.771 137.802C164.324 137.802 164.125 137.583 164.125 137.11V134.694H165.159V133.639H164.125V132.335H162.812Z" fill="white"/>
<path d="M168.426 138.958C169.956 138.958 170.876 137.936 170.876 136.217C170.876 134.513 169.942 133.476 168.426 133.476C166.91 133.476 165.976 134.517 165.976 136.217C165.976 137.931 166.896 138.958 168.426 138.958ZM168.426 137.86C167.722 137.86 167.325 137.258 167.325 136.217C167.325 135.186 167.727 134.579 168.426 134.579C169.121 134.579 169.527 135.186 169.527 136.217C169.527 137.258 169.125 137.86 168.426 137.86Z" fill="white"/>
<path d="M171.95 138.848H173.264V135.807C173.264 135.095 173.638 134.637 174.247 134.637C174.87 134.637 175.163 135.023 175.163 135.764V138.848H176.477V135.448C176.477 134.193 175.881 133.476 174.775 133.476C174.04 133.476 173.534 133.844 173.304 134.436H173.223V133.591H171.95V138.848Z" fill="white"/>
<rect x="62" y="58" width="201" height="8" rx="4" fill="#009990" fill-opacity="0.5"/>
<rect x="62" y="74" width="154" height="8" rx="4" fill="#009990" fill-opacity="0.5"/>
<rect x="84" y="37" width="100" height="12" rx="6" fill="#009990" fill-opacity="0.5"/>
<path d="M0 0V-2H-2V0H0ZM325 0H327V-2H325V0ZM325 180V182H327V180H325ZM0 180H-2V182H0V180ZM0 2H325V-2H0V2ZM323 0V180H327V0H323ZM325 178H0V182H325V178ZM2 180V0H-2V180H2Z" fill="#767676"/>
</svg>
