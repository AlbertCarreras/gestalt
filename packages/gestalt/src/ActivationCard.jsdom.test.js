// @flow strict
import { render } from '@testing-library/react';
import ActivationCard from './ActivationCard.js';

describe('<ActivationCard />', () => {
  it('Pending ActivationCard', () => {
    const { container } = render(
      <ActivationCard
        status="pending"
        statusMessage="Pending"
        title="Claiming your website"
        message="We will notify you via email as soon as your site has been successfully claimed"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Needs attention ActivationCard', () => {
    const { container } = render(
      <ActivationCard
        status="needsAttention"
        statusMessage="Needs attention"
        title="Tag is unhealthy"
        message="Oops! Your tag has errors that need to be fixed"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Not started ActivationCard', () => {
    const { container } = render(
      <ActivationCard
        status="notStarted"
        statusMessage="Not started"
        title="Claim your website"
        message="Claim your website to unlock new features"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('complete ActivationCard', () => {
    const { container } = render(
      <ActivationCard
        status="complete"
        statusMessage="Complete"
        title="Nice work"
        message="You are all done here"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('message + title', () => {
    const { container } = render(
      <ActivationCard
        status="needsAttention"
        statusMessage="Needs attention"
        title="Tag is unhealthy"
        message="Oops! Your tag must be healthy to continue."
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('message + title + link', () => {
    const { container } = render(
      <ActivationCard
        status="needsAttention"
        statusMessage="Needs attention"
        title="Tag is unhealthy"
        message="Oops! Your tag must be healthy to continue."
        link={{
          href: 'https://pinterest.com',
          label: 'Learn more',
          accessibilityLabel: 'Learn more about tags',
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('message + title + link + dismissButton', () => {
    const { container } = render(
      <ActivationCard
        status="needsAttention"
        statusMessage="Needs attention"
        title="Tag is unhealthy"
        message="Oops! Your tag must be healthy to continue."
        link={{
          href: 'https://pinterest.com',
          label: 'Learn more',
          accessibilityLabel: 'Learn more about tags',
        }}
        dismissButton={{
          accessibilityLabel: 'Dismiss card',
          onDismiss: jest.fn(),
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('ActivationCard handles onDismiss callback', () => {
    const mockOnDismiss = jest.fn<[], void>();
    const { getByLabelText } = render(
      <ActivationCard
        status="pending"
        statusMessage="Pending"
        title="Claiming your website"
        message="We will notify you via email as soon as your site has been successfully claimed"
        dismissButton={{
          accessibilityLabel: 'Dismiss card',
          onDismiss: mockOnDismiss,
        }}
      />,
    );
    getByLabelText('Dismiss card').click();
    expect(mockOnDismiss).toHaveBeenCalled();
  });
});
