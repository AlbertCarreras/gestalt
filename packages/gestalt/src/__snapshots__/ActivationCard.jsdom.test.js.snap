// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivationCard /> Needs attention ActivationCard 1`] = `
<div>
  <div
    class="box elevationFloating flexGrow justifyCenter paddingX6 paddingY6 relative rounding4 shadow xsDirectionColumn xsDisplayFlex"
    style="max-width: 400px; height: 100%; width: 100%;"
  >
    <div
      class="box contentCenter xsDisplayFlex"
      style="height: 24px;"
    >
      <div
        class="box marginEnd2"
      >
        <svg
          aria-label="Needs attention"
          class="icon errorIcon iconBlock"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        class="box marginTop0 selfCenter"
      >
        <div
          class="Text fontSize200 defaultText alignStart breakWord fontWeightSemiBold"
        >
          Needs attention
        </div>
      </div>
    </div>
    <div
      class="box marginTop6"
    >
      <div
        class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
      >
        Tag is unhealthy
      </div>
    </div>
    <div
      class="box contentStart flexGrow marginTop2 xsDirectionColumn"
    >
      <div
        class="Text fontSize200 subtleText alignStart breakWord fontWeightNormal"
      >
        Oops! Your tag has errors that need to be fixed
      </div>
    </div>
  </div>
</div>
`;

exports[`<ActivationCard /> Not started ActivationCard 1`] = `
<div>
  <div
    class="box elevationFloating flexGrow justifyCenter paddingX6 paddingY6 relative rounding4 shadow xsDirectionColumn xsDisplayFlex"
    style="max-width: 400px; height: 100%; width: 100%;"
  >
    <div
      class="box contentCenter xsDisplayFlex"
      style="height: 24px;"
    >
      <div
        class="box marginTop1 selfCenter"
      >
        <div
          class="Text fontSize200 subtleText alignStart breakWord fontWeightSemiBold"
        >
          Not started
        </div>
      </div>
    </div>
    <div
      class="box marginTop6"
    >
      <div
        class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
      >
        Claim your website
      </div>
    </div>
    <div
      class="box contentStart flexGrow marginTop2 xsDirectionColumn"
    >
      <div
        class="Text fontSize200 subtleText alignStart breakWord fontWeightNormal"
      >
        Claim your website to unlock new features
      </div>
    </div>
  </div>
</div>
`;

exports[`<ActivationCard /> Pending ActivationCard 1`] = `
<div>
  <div
    class="box elevationFloating flexGrow justifyCenter paddingX6 paddingY6 relative rounding4 shadow xsDirectionColumn xsDisplayFlex"
    style="max-width: 400px; height: 100%; width: 100%;"
  >
    <div
      class="box contentCenter xsDisplayFlex"
      style="height: 24px;"
    >
      <div
        class="box marginEnd2"
      >
        <svg
          aria-label="Pending"
          class="icon subtleIcon iconBlock"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        class="box marginTop0 selfCenter"
      >
        <div
          class="Text fontSize200 defaultText alignStart breakWord fontWeightSemiBold"
        >
          Pending
        </div>
      </div>
    </div>
    <div
      class="box marginTop6"
    >
      <div
        class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
      >
        Claiming your website
      </div>
    </div>
    <div
      class="box contentStart flexGrow marginTop2 xsDirectionColumn"
    >
      <div
        class="Text fontSize200 subtleText alignStart breakWord fontWeightNormal"
      >
        We will notify you via email as soon as your site has been successfully claimed
      </div>
    </div>
  </div>
</div>
`;

exports[`<ActivationCard /> complete ActivationCard 1`] = `
<div>
  <div
    class="box elevationFloating flexGrow justifyCenter paddingX6 paddingY6 relative rounding4 shadow xsDirectionColumn xsDisplayFlex"
    style="max-width: 400px; height: 100%; width: 100%;"
  >
    <div
      class="box xsDisplayFlex"
    >
      <div
        class="box contentCenter xsDisplayFlex"
      >
        <div
          class="box marginEnd4"
        >
          <svg
            aria-label="Complete"
            class="icon successIcon iconBlock"
            height="40"
            role="img"
            viewBox="0 0 24 24"
            width="40"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="box"
        >
          <div
            class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
          >
            Nice work
          </div>
        </div>
        <div
          class="box contentStart flexGrow marginTop2 xsDirectionColumn"
        >
          <div
            class="Text fontSize200 subtleText alignStart breakWord fontWeightNormal"
          >
            You are all done here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ActivationCard /> message + title + link + dismissButton 1`] = `
<div>
  <div
    class="box elevationFloating flexGrow justifyCenter paddingX6 paddingY6 relative rounding4 shadow xsDirectionColumn xsDisplayFlex"
    style="max-width: 400px; height: 100%; width: 100%;"
  >
    <div
      class="box contentCenter xsDisplayFlex"
      style="height: 24px;"
    >
      <div
        class="box marginEnd2"
      >
        <svg
          aria-label="Needs attention"
          class="icon errorIcon iconBlock"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        class="box marginTop0 selfCenter"
      >
        <div
          class="Text fontSize200 defaultText alignStart breakWord fontWeightSemiBold"
        >
          Needs attention
        </div>
      </div>
    </div>
    <div
      class="box marginTop6"
    >
      <div
        class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
      >
        Tag is unhealthy
      </div>
    </div>
    <div
      class="box contentStart flexGrow marginTop2 xsDirectionColumn"
    >
      <div
        class="Text fontSize200 subtleText alignStart breakWord fontWeightNormal"
      >
        Oops! Your tag must be healthy to continue.
      </div>
    </div>
    <div
      class="box"
    >
      <div
        class="box marginEndAuto marginStartAuto marginTop8 paddingX1 pill xsItemsCenter"
      >
        <a
          aria-label="Learn more about tags"
          class="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg gray"
          href="https://pinterest.com"
          rel=""
          tabindex="0"
        >
          <div
            class="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              class="FlexItem"
            >
              <div
                class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Learn more
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="rtlPos"
    >
      <button
        aria-label="Dismiss card"
        class="parentButton"
        tabindex="0"
        type="button"
      >
        <div
          class="button tapTransition enabled"
        >
          <div
            class="pog transparent"
            style="height: 48px; width: 48px;"
          >
            <svg
              aria-hidden="true"
              aria-label=""
              class="icon subtleIcon iconBlock"
              height="16"
              role="img"
              viewBox="0 0 24 24"
              width="16"
            >
              <path
                d="test-file-stub"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`<ActivationCard /> message + title + link 1`] = `
<div>
  <div
    class="box elevationFloating flexGrow justifyCenter paddingX6 paddingY6 relative rounding4 shadow xsDirectionColumn xsDisplayFlex"
    style="max-width: 400px; height: 100%; width: 100%;"
  >
    <div
      class="box contentCenter xsDisplayFlex"
      style="height: 24px;"
    >
      <div
        class="box marginEnd2"
      >
        <svg
          aria-label="Needs attention"
          class="icon errorIcon iconBlock"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        class="box marginTop0 selfCenter"
      >
        <div
          class="Text fontSize200 defaultText alignStart breakWord fontWeightSemiBold"
        >
          Needs attention
        </div>
      </div>
    </div>
    <div
      class="box marginTop6"
    >
      <div
        class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
      >
        Tag is unhealthy
      </div>
    </div>
    <div
      class="box contentStart flexGrow marginTop2 xsDirectionColumn"
    >
      <div
        class="Text fontSize200 subtleText alignStart breakWord fontWeightNormal"
      >
        Oops! Your tag must be healthy to continue.
      </div>
    </div>
    <div
      class="box"
    >
      <div
        class="box marginEndAuto marginStartAuto marginTop8 paddingX1 pill xsItemsCenter"
      >
        <a
          aria-label="Learn more about tags"
          class="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg gray"
          href="https://pinterest.com"
          rel=""
          tabindex="0"
        >
          <div
            class="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              class="FlexItem"
            >
              <div
                class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Learn more
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ActivationCard /> message + title 1`] = `
<div>
  <div
    class="box elevationFloating flexGrow justifyCenter paddingX6 paddingY6 relative rounding4 shadow xsDirectionColumn xsDisplayFlex"
    style="max-width: 400px; height: 100%; width: 100%;"
  >
    <div
      class="box contentCenter xsDisplayFlex"
      style="height: 24px;"
    >
      <div
        class="box marginEnd2"
      >
        <svg
          aria-label="Needs attention"
          class="icon errorIcon iconBlock"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        class="box marginTop0 selfCenter"
      >
        <div
          class="Text fontSize200 defaultText alignStart breakWord fontWeightSemiBold"
        >
          Needs attention
        </div>
      </div>
    </div>
    <div
      class="box marginTop6"
    >
      <div
        class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
      >
        Tag is unhealthy
      </div>
    </div>
    <div
      class="box contentStart flexGrow marginTop2 xsDirectionColumn"
    >
      <div
        class="Text fontSize200 subtleText alignStart breakWord fontWeightNormal"
      >
        Oops! Your tag must be healthy to continue.
      </div>
    </div>
  </div>
</div>
`;
